conda_build/utils.py

```diff
@@ -1631,59 +1631,77 @@ def rm_rf(path):
-        self.max_level = exclusive_maximum
-
-    def filter(self, record):
-        # non-zero return means we log this message
-        return 1 if record.levelno < self.max_level else 0
-
-
-class GreaterThanFilter(logging.Filter):
-    def __init__(self, exclusive_minimum, name=""):
-        super().__init__(name)
-        self.min_level = exclusive_minimum
-
-    def filter(self, record):
-        # non-zero return means we log this message
-        return 1 if record.levelno > self.min_level else 0
-
-
-# unclutter logs - show messages only once
-class DuplicateFilter(logging.Filter):
-    def __init__(self):
-        self.msgs = set()
-
-    def filter(self, record):
-        log = record.msg not in self.msgs
-        self.msgs.add(record.msg)
-        return int(log)
-
+deprecated.constant(
+    "24.5",
+    "24.7",
+    "LessThanFilter",
+    _LessThanFilter,
+    addendum="Use `conda_build.cli.logging.LessThanFilter` instead.",
+)
+deprecated.constant(
+    "24.5",
+    "24.7",
+    "GreaterThanFilter",
+    _GreaterThanFilter,
+    addendum="Use `conda_build.cli.logging.GreaterThanFilter` instead.",
+)
+deprecated.constant(
+    "24.5",
+    "24.7",
+    "DuplicateFilter",
+    _DuplicateFilter,
+    addendum="Use `conda_build.cli.logging.DuplicateFilter` instead.",
+)
 
-dedupe_filter = DuplicateFilter()
-info_debug_stdout_filter = LessThanFilter(logging.WARNING)
-warning_error_stderr_filter = GreaterThanFilter(logging.INFO)
-level_formatter = logging.Formatter("%(levelname)s: %(message)s")
+deprecated.constant(
+    "24.5",
+    "24.7",
+    "dedupe_filter",
+    _dedupe_filter := _DuplicateFilter(),
+    addendum="Use `conda_build.cli.logging.DuplicateFilter()` instead.",
+)
+deprecated.constant(
+    "24.5",
+    "24.7",
+    "info_debug_stdout_filter",
+    _info_debug_stdout_filter := _LessThanFilter(logging.WARNING),
+    addendum="Use `conda_build.cli.logging.LessThanFilter(logging.WARNING)` instead.",
+)
+deprecated.constant(
+    "24.5",
+    "24.7",
+    "warning_error_stderr_filter",
+    _warning_error_stderr_filter := _GreaterThanFilter(logging.INFO),
+    addendum="Use `conda_build.cli.logging.GreaterThanFilter(logging.INFO)` instead.",
+)
+deprecated.constant(
+    "24.5",
+    "24.7",
+    "level_formatter",
+    _level_formatter := logging.Formatter("%(levelname)s: %(message)s"),
+)
 
 # set filelock's logger to only show warnings by default
-logging.getLogger("filelock").setLevel(logging.WARN)
+logging.getLogger("filelock").setLevel(logging.WARNING)
 
 # quiet some of conda's less useful output
-logging.getLogger("conda.core.linked_data").setLevel(logging.WARN)
-logging.getLogger("conda.gateways.disk.delete").setLevel(logging.WARN)
-logging.getLogger("conda.gateways.disk.test").setLevel(logging.WARN)
+logging.getLogger("conda.core.linked_data").setLevel(logging.WARNING)
+logging.getLogger("conda.gateways.disk.delete").setLevel(logging.WARNING)
+logging.getLogger("conda.gateways.disk.test").setLevel(logging.WARNING)
```

Comment:
This is unnecessary since `logging.WARNING` is already the default log level in `conda`.